File: JavaScript/Array/L_averageWaitTime.js
var averageWaitingTime = function (customers) {
    let curr_time = 0;
    let total_wait = 0;
    for (let arrival of customers) {
        if (curr_time <= arrival[0]) {
            total_wait += arrival[1];
            curr_time = arrival[0] + arrival[1];
        } else {
            total_wait += arrival[1] + curr_time - arrival[0];
            curr_time += arrival[1];
        }
    }
    return total_wait / customers.length;
};

File: Python/Arrays/L_averageWaitTime.py
def averageWaitingTime(customers) -> float:
    total_wait = 0
    current_time = 0
    for arrival, ordt in customers:
        if current_time <= arrival:
            total_wait += ordt
            current_time = arrival + ordt
        else:
            total_wait += ordt + current_time - arrival
            current_time += ordt
    return total_wait / len(customers)